### PR TITLE
DOC: added string processing comparison with SAS

### PR DESCRIPTION
**Pull request by natethedrummer**
#13229 

Added string processing section to comparison_with_sas DOC.  Used info from http://blog.dominodatalab.com/using-r-and-python-for-common-sas-functions.  

This completes half of the issue.  Working on R next.

### CODE REVIEW

File: doc/source/comparison_with_sas.rst [hunk 1]
```diff
@@ -357,6 +357,142 @@ takes a list of columns to sort by.
...
+   set tips;
+   put(LENGTHN(time));
+   put(LENGTHC(time));
+   run;
+
+Python determines the length of a character string with the ``len`` function.
+``len`` includes trailing blanks.  Use ``len`` and ``rstrip`` to exclude 
+trailing blanks.
+
+.. code-block:: none
```
Comment (chris-b1) [inline doc/source/comparison_with_sas.rst]: Can you make all the python code segments ipython blocks (like elsewhere in the doc) so the output will show?

**Review by chris-b1 — CHANGES_REQUESTED** [review]
One comment, overall this looks great!

File: doc/source/comparison_with_sas.rst [hunk 2]
```diff
@@ -357,6 +357,142 @@ takes a list of columns to sort by.
    tips = tips.sort_values(['sex', 'total_bill'])
    tips.head()
```
Comment (jreback) [inline doc/source/comparison_with_sas.rst]: can you add a directive here (and other sub-sections if you can), for references e.g. ``_compare_with_sas.string``

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jorisvandenbossche** [comment]
@natethedrummer Thanks!

**Pull request merged**